src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -13,6 +13,8 @@
 
 import java.util.List;
 
+import static io.pinecone.clients.Index.validateListEndpointParameters;
```

Comment:
Since this is used in both places, it should be implemented as a concrete function in `IndexInterface`. You can look at the other validator functions on the interface for examples: https://github.com/pinecone-io/pinecone-java-client/blob/ee8c077d5707a27050705df2f5252d59537a13fb/src/main/java/io/pinecone/commons/IndexInterface.java#L42

Reply:
Done!